tools/src/versioning/android/index.ts

```diff
@@ -432,14 +439,56 @@ async function copyExpoModulesAsync(version: string) {
       pkg.isIncludedInExpoClientOnPlatform('android') &&
       pkg.isVersionableOnPlatform('android')
     ) {
-      await spawnAsync(
-        './android-copy-expo-module.sh',
-        [pkg.packageName, version, path.join(pkg.path, pkg.androidSubdirectory)],
-        {
-          shell: true,
-          cwd: SCRIPT_DIR,
+      const abiVersion = `abi${version.replace(/\./g, '_')}`;
+      const targetDirectory = path.join(ANDROID_DIR, `versioned-abis/expoview-${abiVersion}`);
+      const sourceDirectory = path.join(pkg.path, pkg.androidSubdirectory);
+      const transforms = expoModulesTransforms(pkg.packageName, abiVersion);
+
+      const files = await searchFilesAsync(sourceDirectory, [
+        './src/main/java/**',
+        './src/main/kotlin/**',
+        './src/main/AndroidManifest.xml',
+      ]);
+
+      for (const javaPkg of packagesToKeep) {
+        const javaPkgWithSlash = javaPkg.replace(/\./g, '/');
+        const pathFromPackage = `./src/main/{java,kotlin}/${javaPkgWithSlash}{/**,*}`;
+        for (const file of files) {
+          if (minimatch(file, pathFromPackage)) {
+            files.delete(file);
+            continue;
+          }
         }
+      }
+
+      for (const sourceFile of files) {
+        await copyFileWithTransformsAsync({
+          sourceFile,
+          targetDirectory,
+          sourceDirectory,
+          transforms,
+        });
+      }
+      const temporaryPackageManifestPath = path.join(
+        targetDirectory,
+        'src/main/TemporaryExpoModuleAndroidManifest.xml'
       );
+      const mainManifestPath = path.join(targetDirectory, 'src/main/AndroidManifest.xml');
+      await spawnAsync('java', [
+        '-jar',
+        path.join(SCRIPT_DIR, 'android-manifest-merger-3898d3a.jar'),
+        '--main',
+        mainManifestPath,
+        '--libs',
+        temporaryPackageManifestPath,
+        '--placeholder',
+        'applicationId=${applicationId}',
+        '--out',
+        mainManifestPath,
+        '--log',
+        'WARNING',
+      ]);
+      await fs.remove(temporaryPackageManifestPath);
```

Comment:
I'm not a fan of this solution, but AndroiManifest needs to go through the transform to rename package, so we can't just use manifest from the unversioned directory

Reply:
this step may also be important to merge expo modules' manifest to expoview. e.g. [the one from expo-image-picker](https://github.com/expo/expo/blob/34427758bfce73e53c296f6c8105a34ff7fcdb18/packages/expo-image-picker/android/src/main/AndroidManifest.xml#L16-L24).

i'm working on @shopify/react-native-skia integration and trying to introduce a similar vendoring approach like ios versioning: having a dedicated gradle project for a module. in the future, we can have a versioned expo module with its own build.gradle and AndroidManifest.xml. we can leverage AGP's default manifest merger and deprecate the custom merger here.